### PR TITLE
Facilitating OEM Communication with App Developers

### PR DESCRIPTION
**Pull request by nickschwab**
The goal of this proposal is to provide OEMs and application developers a simple, intuitive, and organized way to communicate with each other for the purpose of establishing a business relationship, discussing the application's journey through certification, and to follow up about any issues discovered during the application's lifecycle.

### CODE REVIEW

**Comment by nickschwab** [comment]
@jordynmackool this proposal is ready for review. Thank you!

File: proposals/NNNN-developer-oem-communication.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+### Trello
+Trello offers an oauth-based API capable of managing boards, lists, cards, and card comments. There are no documented limits to the number of boards, board members, or lists. Similar to GitHub and GitLab, users would be required to sign up for Trello and link their Trello account to the Developer Portal. Members added to a Trello board via API would likely receive an email from Trello with a direct link to the board, along with other Trello-specific emails.
+
+### Open-source Forum Software
+There are various open-source forum (or forum-like) solutions which we could host such as NodeBB, phpBB, myBB, Discourse, etc, but in my investigation I was only able to find one (Discourse) which offers a comprehensive, documented API. These out-of-the-box generic solutions contain many features which we would likely never use (read: overcomplicated; technical debt), are mostly built upon older technology stacks (PHP, MySQL, Apache) than the rest of SDL's infrastructure, and would likely require heavy customization to achieve seamless integration into the Developer Portal.
```
Comment (jordynmackool) [inline proposals/NNNN-developer-oem-communication.md]: Please refrain from using "I", "we" or "my"

There are various open-source forum (or forum-like) solutions **which **the Developer Portal** could** host such as NodeBB, phpBB, myBB, Discourse, etc, **but when investigating there was** **only one (Discourse)** which offers a comprehensive, documented API. These out-of-the-box generic solutions **contains** many features **which would** likely never  **be used** (read: overcomplicated; technical debt), are mostly built upon older technology stacks (PHP, MySQL, Apache) than the rest of SDL's infrastructure, and would likely require heavy customization to achieve seamless integration into the Developer Portal.

File: proposals/NNNN-developer-oem-communication.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+And modifications for existing APIs as follows:
+
+* GET `/application`: to retrieve applications with a negative OEM allowance filter (that is, retrieve SDLC-certified applications which have not been granted access to a specific OEM by the app developer)
+* GET `/vendor`: to retrieve vendors with a negative OEM allowance filter ()
+
+### Data Storage
+The data for these new features would be stored in new relational tables in the existing SHAID Postgres database, as follows:
+* `application_vendor_message`: stores an auto-generated ID, which application the message is regarding, which OEM thread the message is associated with, who the message is from (OEM or app developer ID), the name of the individual who authored the message, the message contents, creation time, deletion time
+* `application_issue`: stores a globally-unique auto-generated ID, an application-specific unique auto-generated ID, which application the issue is regarding, which OEM created the issue, the issue title, description, whether it is accessible by other OEMs, it's opened/closed status, creation time, updated time
```
Comment (jordynmackool) [inline proposals/NNNN-developer-oem-communication.md]: Update to: 
`application_issue`: stores a globally-unique auto-generated ID, an application-specific unique auto-generated ID, which application the issue is regarding, which OEM created the issue, the issue title, description, whether it is accessible by other OEMs, it's opened/closed status, **creation time/date and updated time/date**

File: proposals/NNNN-developer-oem-communication.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* GET `/application/issue/message`: retrieves text-based messages for a specific issue
+
+And modifications for existing APIs as follows:
+
+* GET `/application`: to retrieve applications with a negative OEM allowance filter (that is, retrieve SDLC-certified applications which have not been granted access to a specific OEM by the app developer)
+* GET `/vendor`: to retrieve vendors with a negative OEM allowance filter ()
+
+### Data Storage
+The data for these new features would be stored in new relational tables in the existing SHAID Postgres database, as follows:
+* `application_vendor_message`: stores an auto-generated ID, which application the message is regarding, which OEM thread the message is associated with, who the message is from (OEM or app developer ID), the name of the individual who authored the message, the message contents, creation time, deletion time
```
Comment (jordynmackool) [inline proposals/NNNN-developer-oem-communication.md]: Update: 
`application_vendor_message`: stores an auto-generated ID, which application the message is regarding, which OEM thread the message is associated with, who the message is from (OEM or app developer ID), the name of the individual who authored the message, the message contents, **creation time/date and deletion time/date**

File: proposals/NNNN-developer-oem-communication.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+* `application_issue`: stores a globally-unique auto-generated ID, an application-specific unique auto-generated ID, which application the issue is regarding, which OEM created the issue, the issue title, description, whether it is accessible by other OEMs, it's opened/closed status, creation time, updated time
+* `application_issue_message`: stores an auto-generated ID, which issue ID the message is targeted to, who the message is from (OEM or app developer ID), the message type ('DISCUSSION', 'CLOSED', 'OPENED'), the name of the individual who authored the message, the message contents, creation time, deletion time
```
Comment (jordynmackool) [inline proposals/NNNN-developer-oem-communication.md]: Update to:
* `application_issue_message`: stores an auto-generated ID, which issue ID the message is targeted to, who the message is from (OEM or app developer ID), the message type ('DISCUSSION', 'CLOSED', 'OPENED'), the name of the individual who authored the message, the message contents, **creation time/date, and deletion time/date**

**Comment by nickschwab** [comment]
@jordynmackool Thanks for the feedback. Changes have been made to accommodate.

**Pull request merged**